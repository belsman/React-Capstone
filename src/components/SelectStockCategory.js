import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

const SelectStockCategory = () => {
  const dispatch = useDispatch();
  const currentSelectedFilter = useSelector(state => state.visibilityFilter);

  const handleInputChange = e => dispatch({ type: 'SET_VISIBILITY_FILTER', payload: e.target.value });

  return (
    <select name="stockCategory" id="stockCategory" value={currentSelectedFilter} onChange={handleInputChange}>
      <option value="activeStocks">Active Stocks</option>
      <option value="gainingStocks">Gainers</option>
      <option value="losingStocks">Losers</option>
    </select>
  );
};

export default SelectStockCategory;
