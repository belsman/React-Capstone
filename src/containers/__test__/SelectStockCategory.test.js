import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

/* eslint-disable no-unused-vars */
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';

import SelectStockCategory from '../SelectStockCategory';

jest.mock('react-redux', () => ({
  useDispatch() {
    return {};
  },
  useSelector() {
    return 'gainingStocks';
  },
}));

jest.mock('react-router-dom', () => ({
  useHistory() {
    return [];
  },
}));

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders with with the correct form input data', () => {
  act(() => {
    render(<SelectStockCategory />, container);
  });

  const select = document.querySelector('select');
  expect(select).toHaveLength(3);

  expect(select.value).toBe('gainingStocks');
});

it('should render the expected output', () => {
  act(() => {
    render(<SelectStockCategory />, container);
  });

  expect(
    pretty(container.innerHTML),
  ).toMatchSnapshot();
});
