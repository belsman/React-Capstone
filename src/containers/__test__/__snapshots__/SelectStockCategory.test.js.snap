// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expected output 1`] = `
"<select name=\\"stockCategory\\" id=\\"stockCategory\\">
  <option value=\\"activeStocks\\">Active Stocks</option>
  <option value=\\"gainingStocks\\">Gainers</option>
  <option value=\\"losingStocks\\">Losers</option>
</select>"
`;
